examples/extended_validation/src/tests.rs

```diff
@@ -0,0 +1,89 @@
+use rocket::testing::MockRequest;
+use rocket::http::Method::*;
+use rocket::http::{ContentType, Status};
+
+use super::rocket;
+
+fn test_login(username: &str,
```

Comment:
We can also use the signature:

```rust
fn test_login(..., body: T)
    where T: Into<Option<&'static str>
{
...
}
```

so that we don't have to write `Some` every time we call this function with a `Some`.